### PR TITLE
Rename some of the functions from the scheduler.

### PR DESCRIPTION
**Pull request by floitschG**
The names are probably less familiar, but more consistent:
- FrameCallback: a callback that is relative to the frame and wants the
  frame offset (a duration) as argument.
- addXFrameCallback: adds the given callback to the internal lists/maps.
- scheduleXFrameCallback (currently only X = ""): add the callback, but
  also trigger a new frame.
- handleX: the method that is invoked when the event-loop or the frame
  calls into the scheduler.
- ensureXYZ: ensure that the callback happens.
  Unfortunately there is the ambiguity between a "callback": it can be a
  closure, or the action of doing a callback, so we end up with:
  ensureBeginFrameCallback, and ensureEventLoopCallback, where
  "callback" means the action of being called back.

### CODE REVIEW

**Comment by floitschG** [comment]
RFC: the patch is not yet finished (need to update existing code), but would like to know what you think about the name changes.

File: packages/flutter/lib/src/animation/ticker.dart [hunk 1]
```diff
@@ -74,6 +74,6 @@ class Ticker {
   void _scheduleTick() {
     assert(isTicking);
     assert(_animationId == null);
-    _animationId = scheduler.requestAnimationFrame(_tick);
+    _animationId = scheduler.scheduleFrameCallback(_tick);
```
Comment (abarth) [inline packages/flutter/lib/src/animation/ticker.dart]: If you're in this code anyway, you might want to have this function return an object with a `cancel` method instead of returning an int.  The int return value was just cargo-culted from the web.

File: packages/flutter/lib/src/scheduler/scheduler.dart [hunk 2]
```diff
@@ -18,7 +18,7 @@ double timeDilation = 1.0;
 /// scheduler's epoch. Use timeStamp to determine how far to advance animation
 /// timelines so that all the animations in the system are synchronized to a
 /// common time base.
-typedef void SchedulerCallback(Duration timeStamp);
+typedef void FrameCallback(Duration timeStamp);
```
Comment (abarth) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: We already have a type like this in the engine:

https://github.com/flutter/engine/blob/master/sky/engine/core/dart/window.dart#L8

We could make the one in dart:ui public and re-export it here, although maybe having a fresh declaration here is better for loose coupling.

File: packages/flutter/lib/src/animation/ticker.dart (hunk 1 above)
Comment (floitschG) [inline packages/flutter/lib/src/animation/ticker.dart]: :)
I have another more experimental branch where I return Subscription objects.
Currently the post and persistent callbacks can't even be canceled.

I will see, if I can integrate the Subscription changes already in this PR.

File: packages/flutter/lib/src/scheduler/scheduler.dart [hunk 3]
```diff
@@ -84,102 +84,138 @@ class Priority {
...
       // Note that we inverse the priority.
       return -e1.priority.compareTo(e2.priority);
     }
   );
 
-  /// Wether this scheduler already requested to be woken up as soon as
-  /// possible.
-  bool _wakingNow = false;
+  /// Wether this scheduler already requested to be called from the event loop.
+  bool _isEventLoopCallbackRegistered = false;
```
Comment (abarth) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: `_hasRegisteredAnEventLoopCallback`

The idea is that these names should be "literate", meaning read like english when combined with "this".  For example, "the scheduler has registered an event loop callback" rather than "the scheduler is event loop callback registered".

File: packages/flutter/lib/src/scheduler/scheduler.dart [hunk 4]
```diff
@@ -84,102 +84,138 @@ class Priority {
...
 
-  /// Wether this scheduler already requested to be woken up in the next frame.
-  bool _wakingNextFrame = false;
+  /// Wether this scheduler already requested to be called at the beginning of
+  /// the next frame.
+  bool _isBeginFrameCallbackRegistered = false;
```
Comment (abarth) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: `_hasRegisteredABeginFrameCallback`

File: packages/flutter/lib/src/scheduler/scheduler.dart [hunk 5]
```diff
@@ -84,102 +84,138 @@ class Priority {
...
+  /// registered callback is still executed during the frame. Otherwise,
+  /// the registered callback is executed during the next frame.
+  ///
+  /// The registered callbacks are executed in the order in which they have been
+  /// registered.
+  void addPostFrameCallback(FrameCallback callback) {
     _postFrameCallbacks.add(callback);
   }
 
   bool _inFrame = false;
```
Comment (abarth) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: `_isInFrame`

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/scheduler/scheduler.dart (hunk 3 above)
Comment (floitschG) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: Nice. I now hope, we didn't violate this too often in the core libraries...
done.

File: packages/flutter/lib/src/scheduler/scheduler.dart (hunk 4 above)
Comment (floitschG) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: done.

File: packages/flutter/lib/src/scheduler/scheduler.dart (hunk 5 above)
Comment (floitschG) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: done.

File: packages/flutter/lib/src/animation/ticker.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/animation/ticker.dart]: It's totally fine to do them as separate PRs.

File: packages/flutter/lib/src/scheduler/scheduler.dart [hunk 6]
```diff
@@ -84,102 +84,138 @@ class Priority {
...
 
-  /// Wether this scheduler already requested to be woken up as soon as
-  /// possible.
-  bool _wakingNow = false;
+  /// Wether this scheduler already requested to be called from the event loop.
+  bool _hasRegisteredEventLoopCallback = false;
 
-  /// Wether this scheduler already requested to be woken up in the next frame.
-  bool _wakingNextFrame = false;
+  /// Wether this scheduler already requested to be called at the beginning of
```
Comment (Hixie) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: s/Wether/Whether/

File: packages/flutter/lib/src/scheduler/scheduler.dart [hunk 7]
```diff
@@ -84,102 +84,138 @@ class Priority {
...
       // Note that we inverse the priority.
       return -e1.priority.compareTo(e2.priority);
     }
   );
 
-  /// Wether this scheduler already requested to be woken up as soon as
-  /// possible.
-  bool _wakingNow = false;
+  /// Wether this scheduler already requested to be called from the event loop.
+  bool _hasRegisteredEventLoopCallback = false;
```
Comment (Hixie) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: If I understand the code correctly, I think this should be hasRequestedEventLoopCallback or hasScheduledEventLoopCallback. Registering the callback is what you do when you set the onFoo field to the handleFoo method. Requesting or scheduling one is what you do when you tell the other side to use the callback.

File: packages/flutter/lib/src/scheduler/scheduler.dart [hunk 8]
```diff
@@ -84,102 +84,138 @@ class Priority {
...
 
-  /// Wether this scheduler already requested to be woken up in the next frame.
-  bool _wakingNextFrame = false;
+  /// Wether this scheduler already requested to be called at the beginning of
+  /// the next frame.
+  bool _hasRegisteredABeginFrameCallback = false;
```
Comment (Hixie) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: s/Registered/Requested/ or /Scheduled/ here too. The case here is clear: you set onBeginFrame once (in the constructor); that's where you register the callback. You set this boolean to true after calling `ui.window.scheduleFrame()`, which is scheduling the callback, rather than registering it.

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 9]
```diff
@@ -34,7 +34,8 @@ class WidgetTester {
   void pump([ Duration duration ]) {
     if (duration != null)
       async.elapse(duration);
-    scheduler.beginFrame(new Duration(milliseconds: clock.now().millisecondsSinceEpoch));
+    scheduler.handleBeginFrame(new Duration(
+        milliseconds: clock.now().millisecondsSinceEpoch));
     async.flushMicrotasks();
   }
```
Comment (Hixie) [inline packages/flutter_test/lib/src/widget_tester.dart]: While you're looking at this (but in a separate patch), if you can find a better way to handle the situation here where this test file calls  `ui.window.onBeginFrame = null;`, that would be great. When I added that code I couldn't really see a clean way to configure how the scheduler should be driven. This solution is bad because it is mucking around with a value that really the scheduler thinks it owns.

**Comment by Hixie** [comment]
LGTM also.

File: packages/flutter/lib/src/scheduler/scheduler.dart (hunk 7 above)
Comment (floitschG) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: Thanks.
Done.

File: packages/flutter/lib/src/scheduler/scheduler.dart (hunk 6 above)
Comment (floitschG) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: done.

File: packages/flutter/lib/src/scheduler/scheduler.dart (hunk 8 above)
Comment (floitschG) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: Requested.

File: packages/flutter_test/lib/src/widget_tester.dart (hunk 9 above)
Comment (floitschG) [inline packages/flutter_test/lib/src/widget_tester.dart]: Will do.

File: packages/flutter/lib/src/scheduler/scheduler.dart (hunk 2 above)
Comment (floitschG) [inline packages/flutter/lib/src/scheduler/scheduler.dart]: I will leave it in this CL, but will follow up.
Since onBeginFrame is public, we might as well make its type public too, and then we should probably have only one type.
However, it requires two move the Engine and Flutter forward in sync, so I prefer doing that change in separate PRs.
